src/client.js

```diff
@@ -150,6 +178,61 @@ class PureCloudWebrtcSdk extends WildEmitter {
       }
     });
   }
+
+  _log (level, message, details) {
+    // immediately log it locally
+    this.logger[level](message, details);
+
+    // ex: if level is debug and config is warn, then debug is less than warn, don't push
+    if (LOG_LEVELS.indexOf(level) < LOG_LEVELS.indexOf(this._logLevel)) {
+      return;
+    }
+
+    if (this._optOutOfTelemetry) {
+      return;
+    }
+
+    const log = {
+      clientTime: new Date().toISOString(),
+      clientId: this._clientId,
+      message,
+      details
+    };
+    const logContainer = {
+      topic: `purecloud-webrtc-sdk.${this._clientId}`,
+      level: level.toUpperCase(),
+      message: stringify(log)
+    };
+    this._logBuffer.push(logContainer);
+    this._notifyLogs(); // debounced _sendLogs
+  }
+
+  _notifyLogs () {
+    if (this._logTimer) {
+      clearTimeout(this._logTimer);
+    }
+    this._logTimer = setTimeout(() => {
+      this._sendLogs();
+    }, 1000);
+  }
+
+  _sendLogs () {
+    const traces = this._logBuffer.splice(0, this._logBuffer.lenth);
+    const payload = {
+      app: {
+        appId: 'webrtc-sdk',
+        appVersion: '[AIV]{version}[/AIV]' // injected by webpack auto-inject-version
+      },
+      traces
+    };
+    return requestApi.call(this, '/diagnostics/trace', {
+      method: 'post',
+      contentType: 'application/json; charset=UTF-8',
+      data: JSON.stringify(payload)
+    }).catch(() => {
```

Comment:
are we not going to retry or implement exponential backoff?

Reply:
I thought about that. I think I'd like to do that in another version if necessary. Logs from this shouldn't be critical and should be pretty low in volume.